src/features/api-keys/routes.ts

```diff
@@ -117,3 +123,20 @@ export const createNamedUserApiKey: RequestHandler = async (req, res) => {
 		res.status(500).send(translateError(err));
 	}
 };
+
+const getKeyMetadata = (reqBody: { name: any; description: any }) => {
+	const { name, description } = reqBody;
+
+	if (name != null && !(typeof name === 'string' || name instanceof String)) {
+		throw Error('Key name should be a string value');
```

Comment:
The thrown error should be `BadRequest` so that the correct response code is sent.